libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -321,12 +329,15 @@ private async Task SendOAuthCardAsync(ITurnContext turnContext, IMessageActivity
                                 Title = _settings.Title,
                                 Text = _settings.Text,
                                 Type = ActionTypes.Signin,
+                                Value = signInLink,
                             },
                         },
                     },
                 });
             }
 
+            turnContext?.TurnState?.Add<object>(TurnStateConstants.OAuthLoginTimeoutKey, _settings.Timeout);
```

Comment:
This line seems very complex, with nulls and variations. Please add a detailed comment.